### ISSUE TITLE
Certificate error on Let's Encrypt SSL using curl/wget

### ISSUE THREAD

**Issue by nbyloff** [roots/trellis#539]
I tried to run some commands after updating my domain to HTTPS using the new ansible letsencrypt task. It through an untrusted certificate error so I tried pulling up the domain with wget.

This was the response

`wget www.mydomain.com`

````javascript
ERROR: cannot verify www.mydomain.com's certificate, issued by ‘/C=US/O=Let's Encrypt/CN=Let's Encrypt Authority X3’:
  Unable to locally verify the issuer's authority.
To connect to www.mydomain.com insecurely, use `--no-check-certificate'.
````

But when I run this command against the test domain for letsencrypt.org, I got a successful response

`wget helloworld.letsencrypt.org`

I ran openssl command against my domain as well as the helloworld.letsencrypt.org domains, and got different responses on the certificate chain:

Theirs (only showing chain)

`openssl s_client -host helloworld.letsencrypt.org -port 443`

````javascript
CONNECTED(00000003)
depth=2 C = US, O = IdenTrust, CN = IdenTrust Commercial Root CA 1
verify error:num=20:unable to get local issuer certificate
verify return:0
---
Certificate chain
 0 s:/CN=letsencrypt.org/O=INTERNET SECURITY RESEARCH GROUP/L=Mountain View/ST=California/C=US
   i:/C=US/O=IdenTrust/OU=TrustID Server/CN=TrustID Server CA A52
 1 s:/C=US/O=IdenTrust/OU=TrustID Server/CN=TrustID Server CA A52
   i:/C=US/O=IdenTrust/CN=IdenTrust Commercial Root CA 1
 2 s:/C=US/O=IdenTrust/CN=IdenTrust Commercial Root CA 1
   i:/O=Digital Signature Trust Co./CN=DST Root CA X3
````

And here is my response:

`openssl s_client -host www.mydomain.com -port 443`

````javascript
CONNECTED(00000003)
depth=0 CN = mydomain.com
verify error:num=20:unable to get local issuer certificate
verify return:1
depth=0 CN = mydomain.com
verify error:num=27:certificate not trusted
verify return:1
depth=0 CN = mydomain.com
verify error:num=21:unable to verify the first certificate
verify return:1
---
Certificate chain
 0 s:/CN=mydomain.com
   i:/C=US/O=Let's Encrypt/CN=Let's Encrypt Authority X3
 1 s:/C=US/O=Let's Encrypt/CN=Let's Encrypt Authority X1
   i:/O=Digital Signature Trust Co./CN=DST Root CA X3
---
````

I am not sure if [this issue/discussion](https://github.com/letsencrypt/letsencrypt/issues/2026) would help, but figured it's worth a look.

**Comment by swalkinshaw** [comment]
Going by your last cert chain, it looks like this is the problem: https://github.com/roots/trellis/pull/534

It's been fixed but you'd need to update Trellis and re-issue your SSL certificates.

**Comment by nbyloff** [comment]
Ah ok @swalkinshaw. Can I do that by just deleting the old keys in my `/etc/nginx/ssl/letsencrypt` folder? I just re-provisioned the server with the trellis update on the letsencrypt main.yml file, but still get the error.

**Comment by swalkinshaw** [comment]
Yeah you'd need to manually delete the cert + key. Our role skips re-creating them if they exist since it would exhaust Let's Encrypt quotas.

**Comment by nbyloff** [comment]
@swalkinshaw, Have you seen this error when I deleted the key & cert in that folder? It's on the test nginx.conf role/task. I keep rebooting and trying to reprovision. Then I switched it to http, rebooted, and back to https. Keep getting the error below.

````javascript
SSL: error:0B080074:x509 certificate routines:X509_check_private_key:key values mismatch
````

**Comment by swalkinshaw** [comment]
You might need to delete the CSR in `/var/lib/letsencrypt/csrs` as well. So make sure that's delete, the certificate, the bundled cert file and the key file.

**Comment by nbyloff** [comment]
For anyone that finds this issue, Deleting the files above worked,  but I had to do the following. Let's encrypt ping test kept failing until I rolled the site back to http for some reason.

Set the `wordpress_sites.yml` vars `wp_home` and `wp_siteurl` back to http. I also set ssl enabled: to `false` in the same file. Reprovisioning the production server, rebooted, then changed the above back to true and https, reprovisioned again. 

Original issue gone, as well as my key values mismatch. Thanks!
